pypechain/utilities/abi.py

```diff
@@ -16,33 +16,32 @@
 from pypechain.utilities.types import solidity_to_python_type
 
 
-class Input(NamedTuple):
+class Input(TypedDict):
     """An input of a function or event."""
 
     internalType: str
     name: str
     type: str
-    indexed: bool | None = None
+    indexed: bool | None
 
 
-class Output(NamedTuple):
+class Output(TypedDict):
```

Comment:
just curious why NamedTuple didn't work here?

Reply:
Also why you would choose TypedDict over Dataclass?

Reply:
ah good question, because they didn't play well with the parsers i made that were used to typed dicts that web3.py used.  I actually ended up removing dependencies on these in this PR and used web3.py types instead.  originally i had these in here because web3.py doesn't have 'internalType' on Input and Output, so I made my own.  However, web3.py types are non-inclusive so I can still try to .get() 'internalType' no problem so I decided to just stick with web3.py types.  I'm going to remove these alltogehter.